README.md
# Config Maven Plugin


The `Config Maven Plugin` is a maven plugin which allows to print your project config from properties files.

[![Sonatype Nexus Snapshot](https://img.shields.io/nexus/s/io.github.ololx.moonshine/moonshine-config-maven-plugin?label=Nexus%20Snapshot&server=https%3A%2F%2Fs01.oss.sonatype.org)](https://search.maven.org/search?q=g:%22io.github.ololx.moonshine-config-maven-plugin%22)

[![osslifecycle](https://img.shields.io/osslifecycle/ololx/moonshine-config-maven-plugin?style=flat)](OSSMETADATA) [![last_commit](https://img.shields.io/github/last-commit/ololx/moonshine-config-maven-plugin?style=flat&logo=github)](https://github.com/ololx/moonshine-config-maven-plugin/commits) [![release_date](https://img.shields.io/github/release-date/ololx/moonshine-config-maven-plugin?style=flat&logo=github)](https://github.com/ololx/moonshine-config-maven-plugin/releases) 

[![licence](https://img.shields.io/github/license/ololx/moonshine-config-maven-plugin?style=flat)](LICENCE) [![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-2.1-4baaaa.svg?style=flat)](CODE_OF_CONDUCT.md)

![repo_size](https://img.shields.io/github/repo-size/ololx/moonshine-config-maven-plugin?style=flat&logo=github) ![languages_code_size](https://img.shields.io/github/languages/code-size/ololx/moonshine-config-maven-plugin?style=flat&logo=github) ![languages_count](https://img.shields.io/github/languages/count/ololx/moonshine-config-maven-plugin?style=flat&logo=github) ![languages_top](https://img.shields.io/github/languages/top/ololx/moonshine-config-maven-plugin?style=flat&logo=github)

## 📇 Table of Contents

- [About](#about)
- [Demo](#demo)
- [Features](#feature)
- [Getting Started](#getting-started)
- [Built With](#built-with)
- [Contributing](#contributing)
- [Code of Conduct](#code-of-conduct)
- [Versioning](#versioning)
- [Authors](#authors)
- [Licensing](#licensing)

##  📖 About

The `Config Maven Plugin` is a maven plugin which allows to print your project config from properties files. 

## 📸 Demo

The demonstration shows how `Config Maven Plugin` pront properties from property file..

![The demo of config maven plugin usage](https://github.com/ololx/moonshine-config-maven-plugin/blob/assets/print.gif?raw=true)

## 🎚 Features

- The maven plugin which allows to print your project config from properties files.

### To Do

- soon...

### Changelog

- For more information on a releases, a features and a changes, please read the [changelog](CHANGELOG.md) notes.

## 🚦 Getting Started

soon...

### Cloning

For the cloning this repository to a local machine, just use the follows link:

```https
https://github.com/ololx/moonshine-config-maven-plugin
```

## 🛠 Built With

- **[Maven](https://maven.apache.org/)** - Dependency Management.

## 🎉 Contributing

If you want to contribute this project - you are welcome and have fun.
Please visit the [contributing](CONTRIBUTING.md) section for details on this code of conduct, and the process for submitting pull requests.

## 📝 Code of Conduct

In order to ensure that all is welcoming, please review and abide by the [code of conduct](CODE_OF_CONDUCT.md).

## 🗒 Versioning

For the versioning is used [Semantic Versioning](http://semver.org/). For the versions available, see the [changelog](CHANGELOG.md) or the tags on this repository.

## ©️ Authors

* **Alexander A. Kropotin** - *Initial work* - [ololx](https://github.com/ololx).

## 🔏 Licensing

This project is licensed under the Apache license version 2.0 - see the [lisence](LICENSE) document for details.


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/) and this project adheres to [Semantic Versioning](http://semver.org/).

## [Unreleased] - yyyy-mm-dd

### Added

- The new goal for loading properties.

## [0.0.3] - 2022-08-22

### Added
- Fierst initial public release of the plugin, whicl allows to print properties files.


src/main/java/io/github/ololx/moonshine/config/maven/plugin/FileWalker.java
/**
 * Copyright 2022 the project moonshine-config-maven-plugin authors
 * and the original author or authors annotated by {@author}
 * <p>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.github.ololx.moonshine.config.maven.plugin;

import java.io.File;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

/**
 * project moonshine-config-maven-plugin
 * created 20.08.2022 22:27
 *
 * @author Alexander A. Kropotin
 */
public class FileWalker {

    public List<File> walk(String rootDirectory) {
        File file = new File(rootDirectory);

        if (!file.exists() || !file.isDirectory()) {
            return Collections.emptyList();
        }

        return this.walk(file);
    }

    public List<File> walk(File... files) {
        if (files == null) {
            return Collections.emptyList();
        }

        return this.walk(Arrays.asList(files));
    }

    public List<File> walk(Iterable<File> files) {
        if (files == null) {
            return Collections.emptyList();
        }

        List<File> walked = new ArrayList<>();

        files.forEach(file -> {
            if (file.isDirectory()) {
                walked.addAll(walk(file.listFiles()));
            } else {
                walked.add(file);
            }
        });

        return walked;
    }
}


src/main/java/io/github/ololx/moonshine/config/maven/plugin/PseudoTable.java
package io.github.ololx.moonshine.config.maven.plugin;

import java.util.*;
import java.util.function.Consumer;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class PseudoTable {

    private final BorderFormat topRowBorderFormat;

    private final BorderFormat middleRowBorder;

    private final BorderFormat bottomRowBorder;

    private List<String> header;

    private Map<Integer, List<Object>> body;

    private int bodyRowsCount;

    private List<ColumnFormat> columnFormats = new ArrayList<>();

    public PseudoTable(List<ColumnFormat> columnFormats) {
        this.header = new ArrayList<>();
        this.body = new HashMap<>();
        this.columnFormats = columnFormats;
        this.topRowBorderFormat = new BorderFormat(
                BorderStyle.builder()
                        .base(BorderStyle.BASE)
                        .left(BorderStyle.TOP_LEFT)
                        .right(BorderStyle.TOP_RIGHT)
                        .middle(BorderStyle.TOP_MIDDLE)
                        .build(),
                columnFormats
        );
        this.middleRowBorder = new BorderFormat(
                BorderStyle.builder()
                        .base(BorderStyle.BASE)
                        .left(BorderStyle.MIDDLE_LEFT)
                        .right(BorderStyle.MIDDLE_RIGHT)
                        .middle(BorderStyle.MIDDLE_MIDDLE)
                        .build(),
                columnFormats
        );
        this.bottomRowBorder = new BorderFormat(
                BorderStyle.builder()
                        .base(BorderStyle.BASE)
                        .left(BorderStyle.BOTTOM_LEFT)
                        .right(BorderStyle.BOTTOM_RIGHT)
                        .middle(BorderStyle.BOTTOM_MIDDLE)
                        .build(),
                columnFormats
        );
    }

    public void setHeader(List<String> header) {
        if (header == null) {
            throw new IllegalArgumentException("The header must be not null");
        } else if (header.isEmpty()) {
            throw new IllegalArgumentException("The header must contains at least one column name");
        }

        this.header = header;
    }

    public void setBody(Map<Integer, List<Object>> body) {
        if (body == null) {
            throw new IllegalArgumentException("The body must be not null");
        } else if (body.isEmpty()) {
            throw new IllegalArgumentException("The body must contains at least one row");
        }

        this.body = body;
        this.bodyRowsCount = body.size();
    }

    public void addBodyRow(List<Object> row) {
        if (row == null) {
            throw new IllegalArgumentException("The body row must be not null");
        } else if (row.isEmpty()) {
            throw new IllegalArgumentException("The body row must contains at least one row");
        }

        this.body.put(++bodyRowsCount, row);
    }

    public void print(Consumer<String> out) {
        if (this.header.size() != 2) {
            throw new RuntimeException("The header must contains of two cells");
        }

        out.accept(topRowBorderFormat.format);

        final String headerString = '\u2502' + columnFormats.stream()
                .map(format -> String.format(format.format(), header.get(format.index)) + '\u2502')
                .collect(Collectors.joining());
        out.accept(headerString);

        for (Map.Entry<Integer, List<Object>> rowEntry : this.body.entrySet()) {
            final List<Object> row = rowEntry.getValue();
            if (row.isEmpty()) {
                continue;
            }

            out.accept(middleRowBorder.format);

            final Map<Integer, List<String>> rowCells = reformatRowForColumnWidth(row);
            rowCells.entrySet().forEach(rowCellEntry -> {
                StringBuilder printedRow = new StringBuilder("\u2502");
                List<String> cells = rowCellEntry.getValue();

                columnFormats.forEach(format -> {
                    printedRow.append(String.format(format.format(), cells.get(format.getIndex())));
                    printedRow.append('\u2502');
                });

                out.accept(printedRow.toString());
            });
        }

        out.accept(bottomRowBorder.format);
    }

    private Map<Integer, List<String>> reformatRowForColumnWidth(List<Object> row) {
        final Map<Integer, List<String>> rowCells = new HashMap<>();
        columnFormats.forEach(format -> {
            final List<String> cellRows = getStringOrSplitInArray(String.valueOf(row.get(format.getIndex())), format.getWidth());
            IntStream.range(0, cellRows.size()).forEach(cellRowIterator -> {
                final List<String> cells = rowCells.getOrDefault(cellRowIterator, new ArrayList<>(columnFormats.size()));

                while (cells.size() < format.getIndex()) {
                    cells.add("");
                }

                cells.add(cellRows.get(cellRowIterator));
                rowCells.put(cellRowIterator, cells);
            });
        });

        return rowCells;
    }

    private List<String> getStringOrSplitInArray(String str, int size) {
        return Arrays.stream(str.split("(?<=\\G.{"+size+"})")).collect(Collectors.toList());
    }

    public static class ColumnFormat {

        private final String format;

        private final int index;

        private final int width;

        public ColumnFormat(int index, int width) {
            this.index = index;
            this.width = width;

            this.format = " %-" + width + "s ";
        }

        public int getIndex() {
            return index;
        }

        public int getWidth() {
            return this.width;
        }

        public boolean isBordered() {
            return true;
        }

        public String format() {
            return this.format;
        }
    }

    public static class BorderStyle {

        public static final char TOP_LEFT = '\u250C';

        public static final char TOP_RIGHT = '\u2510';

        public static final char TOP_MIDDLE = '\u252C';

        public static final char MIDDLE_LEFT = '\u251C';

        public static final char MIDDLE_RIGHT = '\u2524';

        public static final char MIDDLE_MIDDLE = '\u253C';

        public static final char BOTTOM_LEFT = '\u2514';

        public static final char BOTTOM_RIGHT = '\u2518';

        public static final char BOTTOM_MIDDLE = '\u2534';

        public static final char BASE = '\u2500';

        private final char left;

        private final char middle;

        private final char right;

        private final char base;

        public BorderStyle(char left, char middle, char right, char base) {
            this.left = left;
            this.middle = middle;
            this.right = right;
            this.base = base;
        }

        public char getBase() {
            return base;
        }

        public char getLeft() {
            return left;
        }

        public char getMiddle() {
            return middle;
        }

        public char getRight() {
            return right;
        }

        public static Builder builder() {
            return new Builder();
        }

        public static class Builder {

            private static final char DEFAULT_STYLE = '*';

            private char left = DEFAULT_STYLE;

            private char middle = DEFAULT_STYLE;

            private char right = DEFAULT_STYLE;

            private char base = DEFAULT_STYLE;

            private Builder() {}

            public Builder base(char base) {
                this.base = base;

                return this;
            }

            public Builder left(char left) {
                this.left = left;

                return this;
            }

            public Builder middle(char middle) {
                this.middle = middle;

                return this;
            }

            public Builder right(char right) {
                this.right = right;

                return this;
            }

            public BorderStyle build() {
                return new BorderStyle(
                        this.left,
                        this.middle,
                        this.right,
                        this.base
                );
            }
        }
    }

    public static class BorderFormat {

        private final String format;

        public BorderFormat(BorderStyle style, Collection<ColumnFormat> columnFormats) {
            this.format = columnFormats.stream().map(columnFormat -> {
                final StringBuilder line = new StringBuilder();
                line.append(
                        IntStream.range(0, columnFormat.getWidth() + 2)
                                .mapToObj(formatPointer -> {
                                    return String.valueOf(style.base);
                                }).collect(Collectors.joining())
                );

                if (columnFormat.getIndex() == 0) {
                    line.insert(0, style.left);
                }

                if (columnFormat.getIndex() == columnFormats.size() - 1) {
                    line.append(style.right);
                } else {
                    line.append(style.middle);
                }

                return line.toString();
            }).collect(Collectors.joining());
        }

        public String format() {
            return this.format;
        }
    }
}


src/main/java/io/github/ololx/moonshine/config/maven/plugin/ConfigPluginMojo.java
/**
 * Copyright 2022 the project moonshine-config-maven-plugin authors
 * and the original author or authors annotated by {@author}
 * <p>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.github.ololx.moonshine.config.maven.plugin;

import org.apache.maven.plugin.AbstractMojo;
import org.apache.maven.plugin.MojoExecutionException;
import org.apache.maven.plugins.annotations.Mojo;

/**
 * project moonshine-config-maven-plugin
 * created 07.08.2022 22:52
 *
 * @author Alexander A. Kropotin
 */
@Mojo( name = "config")
public class ConfigPluginMojo extends AbstractMojo {

    public void execute() throws MojoExecutionException {
        getLog().info( "Hello, config." );
    }
}


src/main/java/io/github/ololx/moonshine/config/maven/plugin/PseudoTwoColumnTable.java
package io.github.ololx.moonshine.config.maven.plugin;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.function.Consumer;
import java.util.stream.Stream;

public class PseudoTwoColumnTable {

    private static final String ROW_FORMAT = '\u2502' + " %-40s " + '\u2502' + " %-60s " + '\u2502';

    private static final String TOP_ROW_BORDER = '\u250C'
            + Stream.iterate(0, i -> i++).limit(42).map(i -> "\u2500").reduce(String::concat).get()
            + "\u252C"
            + Stream.iterate(0, i -> i++).limit(62).map(i -> "\u2500").reduce(String::concat).get()
            + '\u2510';

    private static final String MIDDLE_ROW_BORDER = '\u251C'
            + Stream.iterate(0, i -> i++).limit(42).map(i -> "\u2500").reduce(String::concat).get()
            + "\u253C"
            + Stream.iterate(0, i -> i++).limit(62).map(i -> "\u2500").reduce(String::concat).get()
            + '\u2524';

    private static final String BOTTOM_ROW_BORDER = '\u2514'
            + Stream.iterate(0, i -> i++).limit(42).map(i -> "\u2500").reduce(String::concat).get()
            + "\u2534"
            + Stream.iterate(0, i -> i++).limit(62).map(i -> "\u2500").reduce(String::concat).get()
            + '\u2518';

    private List<String> header;

    private Map<Long, List<Object>> body;

    private long bodyRowsCount;

    public PseudoTwoColumnTable() {
        this.header = new ArrayList<>();
        this.body = new HashMap<>();
    }

    public void setHeader(List<String> header) {
        if (header == null) {
            throw new IllegalArgumentException("The header must be not null");
        } else if (header.isEmpty()) {
            throw new IllegalArgumentException("The header must contains at least one column name");
        }

        this.header = header;
    }

    public void setBody(Map<Long, List<Object>> body) {
        if (body == null) {
            throw new IllegalArgumentException("The body must be not null");
        } else if (body.isEmpty()) {
            throw new IllegalArgumentException("The body must contains at least one row");
        }

        this.body = body;
        this.bodyRowsCount = body.size();
    }

    public void addBodyRow(List<Object> row) {
        if (row == null) {
            throw new IllegalArgumentException("The body row must be not null");
        } else if (row.isEmpty()) {
            throw new IllegalArgumentException("The body row must contains at least one row");
        }

        this.body.put(++bodyRowsCount, row);
    }

    public void print(Consumer<String> out) {
        if (this.header.size() != 2) {
            throw new RuntimeException("The header must contains of two cells");
        }

        out.accept(TOP_ROW_BORDER);

        final String headerString = String.format(ROW_FORMAT, header.get(0), header.get(1));
        out.accept(headerString);

        for (Map.Entry<Long, List<Object>> rowEntry : this.body.entrySet()) {
            final List<Object> row = rowEntry.getValue();
            if (row.size() < 2) {
                continue;
            }

            final String[] rowValues = getStringOrSplitInArray(String.valueOf(row.get(1)), 60);

            out.accept(MIDDLE_ROW_BORDER);

            for (int valuePointer = 0; valuePointer < rowValues.length; valuePointer++) {
                if (valuePointer == (rowValues.length - 1) / 2) {
                    out.accept(String.format(ROW_FORMAT, row.get(0), rowValues[valuePointer]));
                } else {
                    out.accept(String.format(ROW_FORMAT, "", rowValues[valuePointer]));
                }
            }
        }

        out.accept(BOTTOM_ROW_BORDER);
    }

    private String[] getStringOrSplitInArray(String str, int size) {
        return str == null ? new String[0] : str.split("(?<=\\G.{"+size+"})");
    }
}


src/main/java/io/github/ololx/moonshine/config/maven/plugin/ResourcePrintingPlugin.java
/**
 * Copyright 2022 the project moonshine-config-maven-plugin authors
 * and the original author or authors annotated by {@author}
 * <p>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.github.ololx.moonshine.config.maven.plugin;

import org.apache.maven.model.Resource;
import org.apache.maven.plugin.AbstractMojo;
import org.apache.maven.plugin.MojoExecutionException;
import org.apache.maven.plugins.annotations.LifecyclePhase;
import org.apache.maven.plugins.annotations.Mojo;
import org.apache.maven.plugins.annotations.Parameter;

import java.util.List;

/**
 * project moonshine-config-maven-plugin
 * created 07.08.2022 22:52
 *
 * @author Alexander A. Kropotin
 */
@Mojo(
        name = "print",
        defaultPhase = LifecyclePhase.PROCESS_RESOURCES
)
public class ResourcePrintingPlugin extends AbstractMojo {

    /**
     * This element describes all of the classpath resources such as
     * properties files associated with a project. These resources are often
     * included in the final package. The default value is src/main/resources.
     */
    @Parameter(defaultValue = "${project.build.resources/resource*}")
    private List<Resource> mainResources;

    @Override
    public void execute() throws MojoExecutionException {
        getLog().info( "Start to load all resources from '../src/main/resources/..' folders" );
        this.printResourcesPath(mainResources);
    }

    private void printResourcesPath(List<Resource> resources) {
        resources.forEach(resource -> {
            getLog().info(resource.getTargetPath());
        });
    }
}


src/main/java/io/github/ololx/moonshine/config/maven/plugin/ConfigPrintingPlugin.java
/**
 * Copyright 2022 the project moonshine-config-maven-plugin authors
 * and the original author or authors annotated by {@author}
 * <p>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.github.ololx.moonshine.config.maven.plugin;

import org.apache.maven.model.FileSet;
import org.apache.maven.model.Resource;
import org.apache.maven.plugin.AbstractMojo;
import org.apache.maven.plugin.MojoExecutionException;
import org.apache.maven.plugins.annotations.LifecyclePhase;
import org.apache.maven.plugins.annotations.Mojo;
import org.apache.maven.plugins.annotations.Parameter;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.util.*;
import java.util.stream.Collectors;

/**
 * project moonshine-config-maven-plugin
 * created 07.08.2022 22:52
 *
 * @author Alexander A. Kropotin
 */
@Mojo(
        name = "print",
        defaultPhase = LifecyclePhase.PROCESS_RESOURCES
)
public final class ConfigPrintingPlugin extends AbstractMojo {

    private static final List<String> HEADER;

    private static final List<PseudoTable.ColumnFormat> COLUMN_FORMATS;

    private static final Set<String> CONFIG_FILES;

    static {
        HEADER = new ArrayList<String>() {{
            add("PROPERTY NAME");
            add("PROPERTY VALUE");
        }};

        COLUMN_FORMATS = new ArrayList<PseudoTable.ColumnFormat>(){{
            add(new PseudoTable.ColumnFormat(0, 40));
            add(new PseudoTable.ColumnFormat(1, 60));
        }};

        CONFIG_FILES = new HashSet<String>() {{
            add(".properties");
        }};
    }

    @Parameter(
            property = "resources",
            defaultValue = "${project.build.resources}",
            required = true,
            readonly = true
    )
    private final List<String> resources = new ArrayList<>();

    @Parameter(
            property = "testResources",
            defaultValue = "${project.build.testResources}",
            required = true,
            readonly = true
    )
    private final List<String> testResources = new ArrayList<>();

    @Parameter(
            property = "outputDirectory",
            defaultValue = "${project.build.outputDirectory}",
            required = true,
            readonly = true
    )
    private String outputDirectory = "";

    FileWalker walker = new FileWalker();

    public void setResources(List<Resource> resources) {
        this.resources.addAll(
                Objects.requireNonNull(resources).stream()
                        .map(FileSet::getDirectory)
                        .collect(Collectors.toList())
        );
    }

    public void setTestResources(List<Resource> testResources) {
        this.testResources.addAll(
                Objects.requireNonNull(testResources).stream()
                        .map(FileSet::getDirectory)
                        .collect(Collectors.toList())
        );
    }

    public void setOutputDirectory(String outputDirectory) {
        this.outputDirectory = Objects.requireNonNull(outputDirectory);
    }

    @Override
    public void execute() throws MojoExecutionException {
        try {
            this.walkingThroughConfigFiles(resources, "resources");
            this.walkingThroughConfigFiles(testResources, "testResources");
            this.walkingThroughConfigFiles(outputDirectory, "outputDirectory");
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    private void walkingThroughConfigFiles(List<String> resources, String alias)
            throws IOException {
        for (String resource : resources) {
            this.walkingThroughConfigFiles(resource, alias);
        }
    }

    private void walkingThroughConfigFiles(String resourceDir, String alias) throws IOException {
        Map<String, File> configFiles = walker.walk(resourceDir)
                .parallelStream()
                .filter(file -> {
                    return CONFIG_FILES.stream()
                            .anyMatch(type -> file.getName().contains(type));
                })
                .peek(file -> getLog().info("Walk on property file - " + file))
                .collect(Collectors.toMap(File::getName, file -> file));

        for (Map.Entry<String, File> configFile : configFiles.entrySet()) {
            List<String> lines = Files.readAllLines(configFile.getValue().toPath());
            getLog().info("Reading config file - " + configFile.getValue());

            PseudoTable table = new PseudoTable(COLUMN_FORMATS);
            table.setHeader(HEADER);

            for (String line : lines) {
                List<Object> propertyTuples = Arrays.stream(line.split("="))
                        .map(v -> (Object) v).collect(Collectors.toList());
                if (propertyTuples.size() < 2) {
                    continue;
                }
                table.addBodyRow(propertyTuples);
            }

            table.print((row) -> getLog().info(row));
            getLog().info("\n");
        }
    }
}
